### PR TITLE
First version of data-plane swagger spec for Time Series Insights PAYG (Preview) SKU.

### PR DESCRIPTION
**Pull request by dmdenmsft**
First version of data-plane swagger spec for Time Series Insights PAYG (Preview) SKU.

@yeskarthik and @Sandip-Raiyani thank you for majority of the work

### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 1]
```diff
@@ -0,0 +1,2044 @@
...
+          }
+        }
+      ]
+    },
+    "TimeSeriesInstance": {
+      "description": "Time series instances are the time series themselves. In most cases, the deviceId or assetId is the unique identifier of the asset in the environment. Instances have descriptive information associated with them called instance fields. At a minimum, instance fields include hierarchy information. They can also include useful, descriptive data like the manufacturer, operator, or the last service date.",
+      "type": "object",
+      "properties": {
+        "timeSeriesId": {
+          "description": "Time Series ID that uniquely identifies the instance. It matches Time Series ID properties in an environment. Immmutable, never null.",
```
Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]:
```suggestion
          "description": "Time Series ID that uniquely identifies the instance. It matches Time Series ID properties in an environment. Immutable, never null.",
```

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 2]
```diff
@@ -0,0 +1,2044 @@
...
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TsiErrorBody"
+          }
+        }
+      }
+    },
+    "TimeSeriesHierarchyOrError": {
+
```
Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]:
```suggestion
              "description": "Highlighted description of time series instance. May be null.",
```

Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]:
```suggestion
              "description": "Highlighted time series type name that this instance belongs to.",
```

Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]:
```suggestion
          "description": "Language-independent, human-readable string that defines a service-specific error code. This code serves as a more specific indicator for the HTTP error code specified in the response. Can be used to programmatically handle specific error cases.",
```

Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]:
```suggestion
          "description": "Language-independent, human-readable string that defines a service-specific error code. This code serves as a more specific indicator for the HTTP error code specified in the response. Can be used to programmatically handle specific error cases.",
```

**Comment by anuchandy** [comment]
@dmdenmsft, @yeskarthik PR LGTM. As you identified, the 2 flattening specific azure validation errors looks like false positive. We've created an https://github.com/Azure/oav/issues/416 to triage this.

My suggestion here is to not to suppress these warnings, because incase at later point if service introduce a legit nested properties bag then tool won't be able to catch it and this will end up bad customer experience in generated SDK.

**Comment by anuchandy** [comment]
@shahabhijeet @dsgouda FYI - `Time Series Insights` is a new data place RP please onboard to .NET.

**Comment by dsgouda** [comment]
@anuchandy Feel free to approve and merge when ready

**Comment by yeskarthik** [comment]
+ @TheGrungeByte

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 3]
```diff
@@ -0,0 +1,2044 @@
...
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          }
+        }
+      },
+      "required": [
+        "id",
```
[the remark this answers is not in the document]
Comment (yeskarthik) [reply specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: Okay, you are right, will make id not required.

[the remark this answers is not in the document]
Comment (anuchandy) [reply specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: thanks for confirming @yeskarthik, Could you double check any other cases like this exists in the swagger?

[the remark this answers is not in the document]
Comment (yeskarthik) [reply specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: Yes, there was one more case and I have fixed them too.

**Comment by dmdenmsft** [comment]
> @dmdenmsft, @yeskarthik PR LGTM. As you identified, the 2 flattening specific azure validation errors looks like false positive. We've created an [Azure/oav#416](https://github.com/Azure/oav/issues/416) to triage this.
> 
> My suggestion here is to not to suppress these warnings, because incase at later point if service introduce a legit nested properties bag then tool won't be able to catch it and this will end up bad customer experience in generated SDK.

I think it is better to leave the warning - we don't want customers who generate a client see an unnecessary warning while the issue is being fixed. I have added a link to the issue to the suppression instead.

**Comment by anuchandy** [comment]
@dmdenmsft just to clarify - customers of Time Series Insights SDK won't get any warning at all. (whether we keep the suppression or not).

What I mean is - if we keep this suppression we (service team and swagger reviewer) may miss to catch it when there is a legit case (e.g. future update of swagger). Not catching it will end up releasing SDK with bad user experience. Then it's a real problem. By the time customer report it, it will be late, at that point we have to re-release the SDK and ask all users to upgrade to it. All we are trying is avoid any such cases.
Hope it clarifies.

**Comment by dmdenmsft** [comment]
> @dmdenmsft just to clarify - customers of Time Series Insights SDK won't get any warning at all. (whether we keep the suppression or not).
> 
> What I mean is - if we keep this suppression we (service team and swagger reviewer) may miss to catch it when there is a legit case (e.g. future update of swagger). Not catching it will end up releasing SDK with bad user experience. Then it's a real problem. By the time customer report it, it will be late, at that point we have to re-release the SDK and ask all users to upgrade to it. All we are trying is avoid any such cases.
> Hope it clarifies.

I respectfully disagree.

Next person making changes to Swagger and running the checks or our scripts to generate code will see warnings, and will spend countless hours trying to figure out, just like I did.

**CODE MUST BE COMPILING CLEAN EVERY DAY.** Suppression are designed to explain why they don't work **today**.

The bugs may never be fixed, there are always a higher priority issues - the delays should NOT mean that the author should continue to see errors - this is very distracting and time consuming.

If you push for it, I can remove the suppression, but this does show how much these processes, countless arguments and poor tools hurt velocity of the teams.

**Comment by anuchandy** [comment]
@dmdenmsft fair enough :). I will merge this.

**Review by anuchandy — APPROVED** [review]
This service is already released and used by many customers.

**Pull request merged**